lib/trento/clusters.ex

```diff
@@ -211,6 +227,25 @@ defmodule Trento.Clusters do
     end
   end
 
+  @spec get_cluster_primary_node(ClusterReadModel.t()) :: {:ok, map()} | {:error, String.t()}
+  def get_cluster_primary_node(%ClusterReadModel{type: type, details: %{nodes: nodes}})
```

Comment:
comment:
I think we need to change this to `get_cluster_primary_nodes` to return all the primary nodes. Otherwise, `scale out` deployment might miss some scenarios